src/main/java/org/mskcc/kickoff/upload/JiraFileUploader.java

```diff
@@ -74,82 +84,98 @@
     private JiraRestClient restClient;
 
     @Override
-    public void deleteExistingFiles(KickoffRequest request) throws FileDeletionException {
-        String summary = request.getId();
-        Issue issue = getIssue(summary);
-
-        LOGGER.info(String.format("Starting to delete existing manifest files for request: %s from issue: %s in " +
-                "project: %s", request.getId(), summary, projectName));
+    public void upload(String summary, KickoffRequest kickoffRequest) {
 
         try {
-            deleteExistingManifestAttachments(request, issue, summary);
-            validateNoAttachmentExist(request, summary);
+            restClient = getJiraRestClient();
+
+            Optional<String> optionalKey = getIssueKeyFromSummary(summary);
+            if (!optionalKey.isPresent()) {
+                return;
+            }
+
+            String key = optionalKey.get();
+            if (kickoffRequest == null) {
+                addInfoComment(key);
+                transitToBadInputsStatus(key);
+                return;
+            }
+
+            issueStatus = retrieveJiraIssueState(key);
+            if (issueStatus instanceof GenerateFilesStatus) {
```

Comment:
I think you could use validateInputs but in RegenerateFiles it would need to check if files are there instead of throwing exception or jsut do nothing. And in other states maybe it could also do nothing as there is nothing to validate. As issue can be in any state initialy when this code is run so it shoudl be able to work for all so it would be better not to throw exceptions in this case

Reply:
@imprezka88 the issue was addressed by moving the validation logic into issuestatus as mentioned above. let me know what you think.